client-registry-santempi/swarm.sh

```diff
@@ -1,94 +1,122 @@
-    try "docker stack deploy -c $COMPOSE_FILE_PATH/docker-compose-postgres.yml $postgres_cluster_compose_param $postgres_dev_compose_param instant" "Failed to deploy SanteMPI Postgres"
-
-    await_postgres_start
-
-    try "docker stack deploy -c ""$COMPOSE_FILE_PATH""/docker-compose.yml $sante_mpi_dev_compose_param instant" "Failed to deploy SanteMPI"
-  elif [ "$ACTION" == "up" ]; then
-    try "docker stack deploy -c $COMPOSE_FILE_PATH/docker-compose-postgres.yml $postgres_cluster_compose_param $postgres_dev_compose_param instant" "Failed to stand up SanteMPI Postgres"
-
-    await_postgres_start
-
-    try "docker stack deploy -c ""$COMPOSE_FILE_PATH""/docker-compose.yml $sante_mpi_dev_compose_param instant" "Failed to stand up SanteMPI"
-
-    if [[ "$STATEFUL_NODES" == "cluster" ]]; then
-      docker::deploy_sanity santedb-www santedb-mpi santempi-psql-1 santempi-psql-2 santempi-psql-3
-    else
-      docker::deploy_sanity santedb-www santedb-mpi santempi-psql-1
-    fi
-  elif [ "$ACTION" == "down" ]; then
-    try "docker service scale instant_santedb-mpi=0 instant_santedb-www=0 instant_santempi-psql-1=0" "Failed to scale down santeMPI"
-
-    if [ "$STATEFUL_NODES" == "cluster" ]; then
-      try "docker service scale instant_santempi-psql-2=0 instant_santempi-psql-3=0" "Failed to scale down santeMPI postgres replicas"
-    fi
-
-  elif [ "$ACTION" == "destroy" ]; then
-    docker::service_destroy santedb-www
-    docker::service_destroy santedb-mpi
-    docker::service_destroy santempi-psql-1
-    docker::try_remove_volume santedb-data
-    docker::try_remove_volume santempi-psql-1-data
-
-    if [ "${STATEFUL_NODES}" == "cluster" ]; then
-      docker::service_destroy santempi-psql-2
-      docker::service_destroy santempi-psql-3
-      docker::try_remove_volume santempi-psql-2-data
-      docker::try_remove_volume santempi-psql-3-data
-      log warn "Volumes are only deleted on the host on which the command is run. Postgres volumes on other nodes are not deleted"
-    fi
+  if [ "${NODE_MODE}" == "cluster" ]; then
+    postgres_cluster_compose_filename="docker-compose-postgres.cluster.yml"
+  fi
+
+  (
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose-postgres.yml" "$postgres_cluster_compose_filename" "$postgres_dev_compose_filename"
+    docker::deploy_sanity "${postgres_services[@]}"
+
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.yml" "$sante_mpi_dev_compose_filename"
+    docker::deploy_sanity "santedb-mpi" "santedb-www"
+  ) ||
+    {
+      log error "Failed to deploy Client Registry SanteMPI package"
+      exit 1
+    }
+}
+
+function scale_services_down() {
+  for service_name in "${service_names[@]}"; do
+    try \
+      "docker service scale instant_$service_name=0" \
+      catch \
+      "Failed to scale down $service_name"
+  done
+}
+
+function destroy_package() {
+  for service_name in "${service_names[@]}"; do
+    docker::service_destroy "$service_name"
+  done
```

Comment:
Should we refactor this into a seperate function that could be used across all packages ? It seems generic enough.

Reply:
done in another PR :)